org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/services/common/code/AbstractCodeTypeWithGeneric.java

```diff
@@ -443,11 +444,12 @@ protected void loadCodes() {
     // 1a add configured codes
     List<? extends CODE> createdList = interceptCreateCodes();
     if (createdList != null) {
-      for (CODE code : createdList) {
+      visit(createdList, (ICodeVisitor<CODE>) (code, treeLevel) -> {
         allCodesOrdered.add(code);
         idToCodeMap.put(code.getId(), code);
-        codeToParentCodeMap.put(code, null);
-      }
+        codeToParentCodeMap.put(code, Optional.ofNullable(code.getParentCode()).map(ICode::getId).map(idToCodeMap::get).orElse(null));
```

Comment:
to improve readability I'll rather use the following snippet without optional/map:

``codeToParentCodeMap.put(code, code.getParentCode() != null ? idToCodeMap.get(code.getParentCode().getId()) : null);``